lib/global-styles.php

```diff
@@ -56,12 +56,13 @@ function gutenberg_experimental_global_styles_enqueue_assets() {
 
 	$settings = gutenberg_get_default_block_editor_settings();
 	$all      = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data( $settings );
-
 	$stylesheet = gutenberg_experimental_global_styles_get_stylesheet( $all );
 	if ( empty( $stylesheet ) ) {
 		return;
 	}
 
+	do_action( 'gutenberg_experimental_global_styles', $all );
```

Comment:
This action probably needs a better name.

Reply:
We don't want to add specific filters for global styles. Can we use the existing filters for editor and front? Given that the intention is for this SVG to be embedded with the global stylesheet is makes sense that it's called directly by the theme json generator. Example of how we're doing this for other properties https://github.com/WordPress/gutenberg/pull/33812/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R841

Reply:
Can you explore how it'd look like to do this in [the theme json generator instead](https://github.com/WordPress/gutenberg/blob/trunk/lib/class-wp-theme-json-gutenberg.php#L830)? 

We've already added there some rules for the block gap to work, perhaps in can serve of inspiration. I wonder if we can do a similar thing for duotone. Something along these lines:

- while processing the block style nodes, we somehow collect the duotone filters in use
- we add a new step that takes those duotone filters in use and create the necessary SVG

Alternatively, instead of enqueueing the duotones in use, we could enqueue all of them by inspecting the `settings.color.duotones` for core and theme.

Reply:
I did this in b95ebb60b8